### PR TITLE
Dev Ex - Cleanup naming in create mod modal and associated saving code

### PR DESCRIPTION
**Pull request by BLoe**
## What does this PR do?

- Updates a lot of naming in the (now-called) `CreateModModal` and associated saving code
- No intended changes to functionality, purely naming updates

## Checklist

- [ ] Add tests
- [ ] New files added to `src/tsconfig.strictNullChecks.json` (if possible)
- [x] Designate a primary reviewer - @grahamlangford

### CODE REVIEW

**Comment by BLoe** [comment]
This is not intended to merge until after #7299

File: src/pageEditor/hooks/useSaveMod.test.ts [hunk 1]
```diff
@@ -233,3 +234,44 @@ describe("useSaveRecipe", () => {
     });
   });
 });
+
+describe("isModEditable", () => {
+  test("returns true if recipe is in editable packages", () => {
```
Comment (grahamlangford) [inline src/pageEditor/hooks/useSaveMod.test.ts:239]: Why are we using the `recipe` terminology here?

File: src/pageEditor/hooks/useSaveMod.ts [hunk 2]
```diff
@@ -45,31 +42,44 @@ import { reactivateEveryTab } from "@/background/messenger/api";
...
+import type { ModDefinition } from "@/types/modDefinitionTypes";
 
 const { actions: optionsActions } = extensionsSlice;
 
-type RecipeSaver = {
-  save: (recipeId: RegistryId) => Promise<void>;
+// Exported for testing
+export function isModEditable(
+  editablePackages: EditablePackageMetadata[],
+  recipe: ModDefinition,
```
Comment (grahamlangford) [inline src/pageEditor/hooks/useSaveMod.ts:53]: NIT: recipe -> modDefinition

File: src/pageEditor/hooks/useSaveMod.test.ts (hunk 1 above)
Comment (BLoe) [reply src/pageEditor/hooks/useSaveMod.test.ts:239]: Just missed a few spots in comments and test names, will fix

**Review by grahamlangford — APPROVED** [review]

**Pull request merged**